file.c

```diff
@@ -236,6 +237,77 @@ int kibosh_read(const char *path UNUSED, char *buf, size_t size, off_t offset,
     } else {
         if (file->type == KIBOSH_FILE_TYPE_NORMAL) {
             fault = kibosh_fs_check_read_fault(fs, file->path);
+
+            // inject read_corrupt fault
+            if (fault >= CORRUPT_ZERO) {
+                double fraction = 0.0;
+                char *file_type = NULL;
+                struct kibosh_fault_base **iter;
+                struct kibosh_fault_read_corrupt *corrupt_fault;
+                for (iter = fs->faults->list; *iter; iter++) {
+                    if (strcmp((*iter)->type, KIBOSH_FAULT_TYPE_READ_CORRUPT) == 0) {
+                        corrupt_fault = (struct kibosh_fault_read_corrupt*)(*iter);
+                        fraction = corrupt_fault->fraction;
+                        file_type = corrupt_fault->file_type;
+                    }
+                }
+
```

Comment:
For the corruption faults, fraction can be zero. In such case, pos will equal to ret and nothing will be done to the buffer. For the file_type, once the fault is successfully parsed into Kibosh, file_type can never be NULL, it can be "" though.